types/dom-speech-recognition/index.d.ts

```diff
@@ -0,0 +1,120 @@
+interface SpeechRecognition extends EventTarget {
+    continuous: boolean;
+    grammars: SpeechGrammarList;
+    interimResults: boolean;
+    lang: string;
+    maxAlternatives: number;
+    onaudioend: ((this: SpeechRecognition, ev: Event) => any) | null;
+    onaudiostart: ((this: SpeechRecognition, ev: Event) => any) | null;
+    onend: ((this: SpeechRecognition, ev: Event) => any) | null;
+    onerror: ((this: SpeechRecognition, ev: SpeechRecognitionErrorEvent) => any) | null;
+    onnomatch: ((this: SpeechRecognition, ev: SpeechRecognitionEvent) => any) | null;
+    onresult: ((this: SpeechRecognition, ev: SpeechRecognitionEvent) => any) | null;
+    onsoundend: ((this: SpeechRecognition, ev: Event) => any) | null;
+    onsoundstart: ((this: SpeechRecognition, ev: Event) => any) | null;
+    onspeechend: ((this: SpeechRecognition, ev: Event) => any) | null;
+    onspeechstart: ((this: SpeechRecognition, ev: Event) => any) | null;
+    onstart: ((this: SpeechRecognition, ev: Event) => any) | null;
+    abort(): void;
+    start(): void;
+    stop(): void;
+    addEventListener<K extends keyof SpeechRecognitionEventMap>(
+        type: K,
+        listener: (this: SpeechRecognition, ev: SpeechRecognitionEventMap[K]) => any,
+        options?: boolean | AddEventListenerOptions,
+    ): void;
+    addEventListener(
+        type: string,
+        listener: EventListenerOrEventListenerObject,
+        options?: boolean | AddEventListenerOptions,
+    ): void;
+    removeEventListener<K extends keyof SpeechRecognitionEventMap>(
+        type: K,
+        listener: (this: SpeechRecognition, ev: SpeechRecognitionEventMap[K]) => any,
+        options?: boolean | EventListenerOptions,
+    ): void;
+    removeEventListener(
+        type: string,
+        listener: EventListenerOrEventListenerObject,
+        options?: boolean | EventListenerOptions,
+    ): void;
+}
+
+declare var SpeechRecognition: { prototype: SpeechRecognition; new (): SpeechRecognition };
+
+// https://wicg.github.io/speech-api/#speechrecognitionevent
+interface SpeechRecognitionEventInit extends EventInit {
+    resultIndex?: number;
+    results: SpeechRecognitionResultList;
+}
+
+// https://wicg.github.io/speech-api/#dictdef-speechrecognitioneventinit
+interface SpeechRecognitionEvent extends Event {
+    readonly resultIndex: number;
+    readonly results: SpeechRecognitionResultList;
+}
+
+declare var SpeechRecognitionEvent: {
+    prototype: SpeechRecognitionEvent;
+    new (type: string, eventInitDict: SpeechRecognitionEventInit): SpeechRecognitionEvent;
+};
+
+// https://wicg.github.io/speech-api/#enumdef-speechrecognitionerrorcode
+type SpeechRecognitionErrorCode =
+    | 'aborted'
+    | 'audio-capture'
+    | 'bad-grammar'
+    | 'language-not-supported'
+    | 'network'
+    | 'no-speech'
+    | 'not-allowed'
+    | 'service-not-allowed';
+
+// https://wicg.github.io/speech-api/#speechrecognitionerrorevent
+interface SpeechRecognitionErrorEvent extends Event {
```

Comment:
I think we're missing the `declare var` and init dict for the SpeechRecognitionErrorEvent.

Reply:
Nice catch. Thank you :) fixed.

Reply:
I also added some more test to cover new types.